multivariate_grouped_data_plot.Rmd

```diff
@@ -0,0 +1,825 @@
+# How to plot grouped data for multivariate (in Chinese)
+
+Yihui Hu, Liuxin Chen
+
+原始网页 Original website：http://www.sthda.com/english/articles/32-r-graphics-essentials/132-plot-grouped-data-box-plot-bar-plot-and-more/#grouped-continuous-variables
+
+```{r, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+
+
+## 1 数据准备
+
+加载所需的软件包并将主题函数theme_pubclean()设置为默认主题：
+
+```{r}
+library(dplyr) 
+library(ggplot2)
+library(ggpubr)
+theme_set(theme_pubclean())
+```
+
+  
+  
+
+## 2 分类变量分组制图
+
+绘图类型：分类变量频率的分组条形图。
+
+关键函数：geom_bar()。
+
+演示数据集：dimonds[ggplot2]。 
+
+演示示例中使用的类别变量为：
+
+- cut：切工的钻石质量（一般，良好，非常好，优质，理想）。
+
+- color：钻石色，从J（最差）到D（最佳）。
+
+在演示示例中，我们将仅绘制数据的子集（颜色J和D）。 
+
+步骤如下：
+
+a.过滤数据以仅保留颜色在其中的菱形（“ J”，“ D”）
+
+b.按切割质量和钻石颜色对数据进行分组
+
+c.按组计算频率
+
+d.创建条形图
+
+### 2.1 筛选特定分组数据及计算频率
+
+```{r}
+df <- diamonds %>%
+  filter(color %in% c("J", "D")) %>%
+  group_by(cut, color) %>%
+  summarise(counts = n()) 
+head(df, 4)
+```
+
+### 2.2 分组频率柱状图
+
+关键函数：geom_bar()。
+
+主要参数：stat ="identity"。
+
+使用scale_color_manual()和scale_fill_manual()函数手动设置条形边框的线条颜色和区域填充颜色。
+
+```{r}
+# 堆积条形图  y = counts，x = cut，同时柱颜色由变量“color”决定
+# position = position_stack() 
+ggplot(df, aes(x = cut, y = counts)) +
+  geom_bar(
+    aes(color = color, fill = color),
+    stat = "identity", position = position_stack()
+    ) +
+  scale_color_manual(values = c("#0073C2FF", "#EFC000FF"))+
+  scale_fill_manual(values = c("#0073C2FF", "#EFC000FF"))
+# position = position_dodge() 
+p <- ggplot(df, aes(x = cut, y = counts)) +
+  geom_bar(
+    aes(color = color, fill = color),
+    stat = "identity", position = position_dodge(0.8),
+    width = 0.7
+    ) +
+  scale_color_manual(values = c("#0073C2FF", "#EFC000FF"))+
+  scale_fill_manual(values = c("#0073C2FF", "#EFC000FF"))
+p
+```
+
+另外，可以使用ggpubr软件包轻松创建点图。
+
+```{r}
+ggdotchart(df, x = "cut", y ="counts",
+           color = "color", palette = "jco", size = 3, 
+           add = "segment", 
+           add.params = list(color = "lightgray", size = 1.5),
+           position = position_dodge(0.3),
+           ggtheme = theme_pubclean()
+           )
+```
+
+### 2.3 加入数据标签（并列柱状图）
+
+```{r}
+p + geom_text(
+  aes(label = counts, group = color), 
+  position = position_dodge(0.8),
+  vjust = -0.3, size = 3.5
+)
+```
+
+### 2.4 加入数据标签（堆积柱状图）
+
+- 按cut和color列对数据进行排序。当position_stack()反转组顺序时，颜色列应按降序排序。
+
+- 计算每个切割类别的累计计数总和。用作标签的y坐标。要将标签放在条形图的中间，我们将使用cumsum（counts）-0.5 *计数。
+
+- 创建条形图并添加标签。
+
+```{r}
+df <- df %>%
+  arrange(cut, desc(color)) %>%
+  mutate(lab_ypos = cumsum(counts) - 0.5 * counts) 
+head(df, 4)
+```
+```{r}
+ggplot(df, aes(x = cut, y = counts)) +
+  geom_bar(aes(color = color, fill = color), stat = "identity") +
+  geom_text(
+    aes(y = lab_ypos, label = counts, group = color),
+    color = "white"
+  ) + 
+  scale_color_manual(values = c("#0073C2FF", "#EFC000FF"))+
+  scale_fill_manual(values = c("#0073C2FF", "#EFC000FF")) 
+```
+
+### 2.5 其他类型柱状图
```

Comment:
Hashtag headers should not be followed by numbers since the hashtags will create numbered subheadings.